### ISSUE TITLE
Onnx Export doesn't acknowledge dynamic_dict for 2D Tensors (Cannot use dynamic dimensions for Points and Labels in Sam)

### ISSUE THREAD

**Issue by FabianSchuetze** [pytorch/pytorch#151430]
### 🐛 Describe the bug

The following code shows that torch.onnx.export doesn't produce a graph with variable input sizes for 2D tensors:
```
import torch


class Module(torch.nn.Module):

    def forward(self, points, labels, image):
        return points


if __name__ == "__main__":
    model = Module()
    image = torch.rand(1, 3, 100, 200)
    points = torch.tensor([[20, 10]])
    labels = torch.tensor([[1]])
    input_names = ["points", "labels", "image"]
    inp_args = (points, labels, image)
    dynamic_dict = {
        "points": {0: "axis_0"},
        "labels": {0: "axis_0"},
        "image": {2: "axis_2", 3: "axis_3"},
    }
    onnx_model = torch.onnx.export(
        model,
        inp_args,
        "/tmp/model.onnx",
        dynamo=True,
        report=True,
        input_names=input_names,
        dynamic_shapes=dynamic_dict,
    )
```
The output graph shows:
```
In [4]: onnx_model
Out[4]: 
ONNXProgram(
    model=
        <
            ir_version=10,
            opset_imports={'pkg.onnxscript.torch_lib.common': 1, '': 18},
            producer_name='pytorch',
            producer_version='2.8.0.dev20250409+cu118',
            domain=None,
            model_version=None,
        >
        graph(
            name=main_graph,
            inputs=(
                %"points"<INT64,[1,2]>,
                %"labels"<INT64,[1,1]>,
                %"image"<FLOAT,[1,3,axis_2,axis_3]>
            ),
            outputs=(
                %"points"<INT64,[1,2]>
            ),
        ) {

            return %"points"<INT64,[1,2]>
        }


    ,
    exported_program=
        ExportedProgram:
            class GraphModule(torch.nn.Module):
                def forward(self, points: "i64[1, 2]", labels: "i64[1, 1]", image: "f32[1, 3, s48, s41]"):
                    return (points,)
            
        Graph signature: 
            # inputs
            points: USER_INPUT
            labels: USER_INPUT
            image: USER_INPUT
    
            # outputs
            points: USER_OUTPUT
    
        Range constraints: {s48: VR[2, int_oo], s41: VR[2, int_oo]}

)
```
I am surprised that dim 2 and 3 or the `image` argument are dynamic, but not dim 0 or the `points` and `labels` argument. What can I do to use a dynamic number of points and labels? 

The example above is a MWE covering problems encountered while exporting SAM2. The output size is a function of the input image size. 

### Versions

Collecting environment information...
/home/fabian/.local/lib/python3.12/site-packages/torch/cuda/__init__.py:174: UserWarning: CUDA initialization: CUDA unknown error - this may be due to an incorrectly set up environment, e.g. changing env variable CUDA_VISIBLE_DEVICES after program start. Setting the available devices to be zero. (Triggered internally at /pytorch/c10/cuda/CUDAFunctions.cpp:109.)
  return torch._C._cuda_getDeviceCount() > 0
PyTorch version: 2.8.0.dev20250409+cu118
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.2 LTS (x86_64)
GCC version: (Ubuntu 14.2.0-4ubuntu2~24.04) 14.2.0
Clang version: 19.1.1 (1ubuntu1~24.04.2)
CMake version: version 3.28.3
Libc version: glibc-2.39

Python version: 3.12.3 (main, Feb  4 2025, 14:48:35) [GCC 13.3.0] (64-bit runtime)
Python platform: Linux-6.11.0-21-generic-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: 12.0.140
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: GPU 0: NVIDIA RTX 500 Ada Generation Laptop GPU
Nvidia driver version: 550.120
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        46 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               22
On-line CPU(s) list:                  0-21
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Core(TM) Ultra 7 155H
CPU family:                           6
Model:                                170
Thread(s) per core:                   2
Core(s) per socket:                   16
Socket(s):                            1
Stepping:                             4
CPU(s) scaling MHz:                   25%
CPU max MHz:                          4800.0000
CPU min MHz:                          400.0000
BogoMIPS:                             5990.40
Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb intel_ppin ssbd ibrs ibpb stibp ibrs_enhanced tpr_shadow flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid rdseed adx smap clflushopt clwb intel_pt sha_ni xsaveopt xsavec xgetbv1 xsaves split_lock_detect user_shstk avx_vnni dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp hwp_pkg_req hfi vnmi umip pku ospke waitpkg gfni vaes vpclmulqdq rdpid bus_lock_detect movdiri movdir64b fsrm md_clear serialize arch_lbr ibt flush_l1d arch_capabilities
Virtualization:                       VT-x
L1d cache:                            544 KiB (14 instances)
L1i cache:                            896 KiB (14 instances)
L2 cache:                             18 MiB (9 instances)
L3 cache:                             24 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-21
Vulnerability Gather data sampling:   Not affected
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Not affected
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS Not affected; BHI BHI_DIS_S
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] fast_pytorch_kmeans==0.2.2
[pip3] flake8==7.1.2
[pip3] mypy==1.15.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==1.26.4
[pip3] nvidia-cublas-cu11==11.11.3.6
[pip3] nvidia-cublas-cu12==12.6.4.1
[pip3] nvidia-cuda-cupti-cu11==11.8.87
[pip3] nvidia-cuda-cupti-cu12==12.6.80
[pip3] nvidia-cuda-nvrtc-cu11==11.8.89
[pip3] nvidia-cuda-nvrtc-cu12==12.6.77
[pip3] nvidia-cuda-runtime-cu11==11.8.89
[pip3] nvidia-cuda-runtime-cu12==12.6.77
[pip3] nvidia-cudnn-cu11==9.1.0.70
[pip3] nvidia-cudnn-cu12==9.5.1.17
[pip3] nvidia-cufft-cu11==10.9.0.58
[pip3] nvidia-cufft-cu12==11.3.0.4
[pip3] nvidia-curand-cu11==10.3.0.86
[pip3] nvidia-curand-cu12==10.3.7.77
[pip3] nvidia-cusolver-cu11==11.4.1.48
[pip3] nvidia-cusolver-cu12==11.7.1.2
[pip3] nvidia-cusparse-cu11==11.7.5.86
[pip3] nvidia-cusparse-cu12==12.5.4.2
[pip3] nvidia-cusparselt-cu12==0.6.3
[pip3] nvidia-nccl-cu11==2.21.5
[pip3] nvidia-nccl-cu12==2.26.2
[pip3] nvidia-nvjitlink-cu12==12.6.85
[pip3] nvidia-nvtx-cu11==11.8.86
[pip3] nvidia-nvtx-cu12==12.6.77
[pip3] onnx==1.17.0
[pip3] onnxruntime==1.20.1
[pip3] onnxscript==0.2.2
[pip3] pytorch-triton==3.3.0+git96316ce5
[pip3] torch==2.8.0.dev20250409+cu118
[pip3] torchprofile==0.0.4
[pip3] torchvision==0.22.0.dev20250410+cu118
[pip3] triton==3.2.0
[pip3] types-flake8-2020==1.8
[pip3] types-flake8-bugbear==23.9.16
[pip3] types-flake8-builtins==2.2
[pip3] types-flake8-docstrings==1.7
[pip3] types-flake8-plugin-utils==1.3
[pip3] types-flake8-rst-docstrings==0.3
[pip3] types-flake8-simplify==0.21
[pip3] types-flake8-typing-imports==1.15
[pip3] types-mypy-extensions==1.0
[conda] Could not collect

cc @chauhang @penguinwu

**Comment by justinchuby** [comment]
Please try using a dim size that is >=2 (not 0 or 1) in the example input: [The 0/1 specialization problem in Pt2 Export](https://docs.google.com/document/d/16VPOa3d-Liikf48teAOmxLc92rgvJdfosIy-yoT38Io/edit?tab=t.0#heading=h.ez923tomjvyk)

**Comment by FabianSchuetze** [comment]
Thanks @justinchuby  , that addresses the issue for the MWE above. Would a warning for the conversion maybe be useful?
